src/common/Executor.cpp

```diff
@@ -108,6 +119,99 @@ Executor::Executor(int argc, char** argv)
+  adiak::value("rajaperf_compiler", cc.adiak_rajaperf_compiler);
+  adiak::value("compiler_version", cc.adiak_compiler_version);
+
+  auto tokens = split(cc.adiak_rajaperf_compiler, "/");
+  string compiler_exec = tokens.back();
+  adiak::catstring compiler = compiler_exec + "-" + std::string(cc.adiak_compiler_version);
+  cout << "Compiler: " << (string)compiler << "\n";
+  adiak::value("compiler", compiler);
+  auto tsize = tokens.size();
+  if (tsize >= 3) {
+    // pickup path version <compiler-version-hash|date>/bin/exec
+    string path_version = tokens[tsize-3];
+    auto s = split(path_version,"-");
+    if (s.size() >= 2) {
+      adiak::path path_version_short = s[0] + "-" + s[1];
+      adiak::value("Compiler_path_version", (adiak::catstring)path_version_short);
+    } 
+  }
+
+  if (cc.adiak_cmake_exe_linker_flags.size() > 0) {
+    adiak::value("cmake_exe_linker_flags", cc.adiak_cmake_exe_linker_flags);
+  }
+  if (cc.adiak_rajaperf_compiler_options.size() > 0) {
+    adiak::value("rajaperf_compiler_options", cc.adiak_rajaperf_compiler_options);
+  }
+  if (std::string(cc.adiak_cuda_compiler_version).size() > 0) {
+    adiak::value("cuda_compiler_version", cc.adiak_cuda_compiler_version);
+  }
+  if (strlen(cc.adiak_gpu_targets) > 0) {
+    adiak::value("gpu_targets", cc.adiak_gpu_targets);
+  }
+  if (strlen(cc.adiak_cmake_hip_architectures) > 0) {
+    adiak::value("cmake_hip_architectures", cc.adiak_cmake_hip_architectures);
+  }
+  if (cc.adiak_gpu_targets_block_sizes.size() > 0) {
+    adiak::value("gpu_targets_block_sizes", cc.adiak_gpu_targets_block_sizes);
+  }
+  if (cc.adiak_raja_hipcc_flags.size() > 0) {
+    adiak::value("raja_hipcc_flags", cc.adiak_raja_hipcc_flags);
+  }
+  if (std::string(cc.adiak_mpi_cxx_compiler).size() > 0) {
+    adiak::value("mpi_cxx_compiler", cc.adiak_mpi_cxx_compiler);
+  }
+  if (std::string(cc.adiak_systype_build).size() > 0) {
+    adiak::value("systype_build", cc.adiak_systype_build);
+  }
+  if (std::string(cc.adiak_machine_build).size() > 0) {
+    adiak::value("machine_build", cc.adiak_machine_build);
+  }
+
+  adiak::value("SizeMeaning",(adiak::catstring)run_params.SizeMeaningToStr(run_params.getSizeMeaning()));
+  if (run_params.getSizeMeaning() == RunParams::SizeMeaning::Factor) {
+    adiak::value("ProblemSizeRunParam",(uint)run_params.getSizeFactor());
+  } else if (run_params.getSizeMeaning() == RunParams::SizeMeaning::Direct) {
+    adiak::value("ProblemSizeRunParam",(uint)run_params.getSize());
+  }
+
+  if (run_params.numValidGPUBlockSize() > 0) {
+    adiak::value("runtime_gpu_block_sizes", run_params.getGPUBlockSizeInput());
+  }
+
+  // Openmp section
+#if defined(_OPENMP)
+  std::string strval = "";
+  std::string test = std::to_string(_OPENMP);
+
+  std::unordered_map<unsigned,std::string> map{
+    {200505,"2.5"},{200805,"3.0"},{201107,"3.1"},{201307,"4.0"},{201511,"4.5"},{201611,"4.5"},{201811,"5.0"},{202011,"5.1"},{202111,"5.2"}};
+
+  try {
```

Comment:
Use find instead of try catch.

Reply:
[962334b](https://github.com/LLNL/RAJAPerf/pull/352/commits/962334b53464aa150e3a9ac9c44343071430943c)